src/components/draw/DrawStyleDirective.js

```diff
@@ -15,16 +15,46 @@ goog.require('ga_window_service');
   module.directive('gaDrawStyle', function($document, $window, $translate,
       gaGlobalOptions, gaStyleFactory, gaUrlUtils, gaWindow) {
 
+    var findCategoryBySource = function(src, categories) {
+      for (var i = 0; i < categories.length; i++) {
+        var c = categories[i];
+        var regex = (c.type === 'img') ?
+          new RegExp('^(.*)' + c.id) :
+          new RegExp(c.regex);
+        if (regex.test(src)) {
+          return c;
+        }
+      }
+      console.error('No category found for source ' + src);
+    };
+
+    var getCategoryById = function(id, categories) {
+      for (var i = 0; i < categories.length; i++) {
+        var c = categories[i];
+        for (var j = 0; j < c.icons.length; j++) {
+          var icon = c.icons[j];
+          if (icon.id === id) {
+            return c;
+          }
+        }
+      }
+      console.error('No category found for id ' + id);
+    }
+
     // Find the corresponding style
-    var findIcon = function(olIcon, icons) {
+    var findIcon = function(olIcon, category, standardIcon) {
```

Comment:
Usually we use `default` key word instead of `standard`

Reply:
You don't need to pass the default icon see other comments l112 . It simplify a bit this function.